ocaml/xapi/xapi_host.ml

```diff
@@ -3055,18 +3060,24 @@ let try_restart_device_models_for_recommended_guidances ~__context ~host =
       None
 
 let apply_recommended_guidances ~__context ~self =
+  (* This function runs on master host *)
+  Helpers.assert_we_are_master ~__context ;
   try
     let open Updateinfo in
     Db.Host.get_recommended_guidances ~__context ~self |> function
     | [] ->
         try_restart_device_models_for_recommended_guidances ~__context
           ~host:self
     | [`reboot_host] ->
-        reboot ~__context ~host:self
+        Helpers.call_api_functions ~__context (fun rpc session_id ->
```

Comment:
It might be worth emulating the do_host_op_on: if the coordinator needs to be restarted it would be nice if it used Local.Host.reboot instead to avoid creating a new connection

Reply:
Hi @psafont , I tried a fix with: [c71c4bc](https://github.com/xapi-project/xen-api/pull/4991/commits/c71c4bc51a7fb8cf6109efbcbbe6026a32f223b1), which "copies" the way "Message_forwarding" handles requests. 
But I still have a question, I greped xen-api code, seems almost all requests to other host is handled with "call_api_functions", while only few special requests, like "hello", "ping" is using the "do_op_on", is there some reason for this? Please help to have a review, thank you!

Reply:
changed back to use call_api_functions instead of do_op_on with [fa8f1b9](https://github.com/xapi-project/xen-api/pull/4991/commits/fa8f1b96fb9ec1df818263b41fe49b1ece8277f5), as the discussion here: https://citrix.slack.com/archives/C7EPZ2JR3/p1683623556253019?thread_ts=1683535636.421609&cid=C7EPZ2JR3